src/core/utils/Ledger.ts

```diff
@@ -0,0 +1,236 @@
+        const signature = h.slice(0, 128);
+        const payload = rawPayload.slice(0, 16) + signature + signerPublicKey + rawPayload.slice(16 + 128 + 64, rawPayload.length);
+        const generationHashBytes = Array.from(Convert.hexToUint8(networkGenerationHash));
+        const transactionHash = Transaction.createTransactionHash(payload, generationHashBytes);
+        const signedTransaction = new SignedTransaction(
+            payload,
+            transactionHash,
+            signerPublicKey,
+            transaction.type,
+            transaction.networkType,
+        );
+        return signedTransaction;
+    }
+
+    /**
+     * sign a Symbol Cosignature transaction with a given BIP 44 path
+     *
+     * @param path a path in BIP 44 format
+     * @param cosignatureTransaction a cosinature transaction needs to be signed
+     * @param signerPublicKey the public key of signer
+     * @return a Signed Cosignature Transaction
+     */
+    public async signCosignatureTransaction(path: string, cosignatureTransaction: AggregateTransaction, signerPublicKey: string) {
+        const rawPayload = cosignatureTransaction.serialize();
+        const signingBytes = cosignatureTransaction.transactionInfo.hash + rawPayload.slice(216);
+        const rawTx = Buffer.from(signingBytes, 'hex');
+        const response = await this.ledgerMessageHandler(path, rawTx);
+        // Response from Ledger
+        const h = response.toString('hex');
+        const signature = h.slice(0, 128);
+        const cosignatureSignedTransaction = new CosignatureSignedTransaction(
+            cosignatureTransaction.transactionInfo.hash,
+            signature,
+            signerPublicKey,
+        );
+        return cosignatureSignedTransaction;
+    }
+    /**
+     * handle sending and receiving packages between Ledger and Wallet
+     * @param path a path in BIP 44 format
+     * @param rawTx a raw payload transaction hex string
+     * @returns respond package from Ledger
+     */
+    private async ledgerMessageHandler(path: string, rawTx: Buffer) {
+        const TX_INS_FIELD = 0x04;
+        const MAX_CHUNK_SIZE = 255;
+        const CONTINUE_SENDING = '0x9000';
+
+        const chainCode = false;
+
+        const curveMask = 0x80; // use Curve25519
+        const bipPath = BIPPath.fromString(path).toPathArray();
+        const apduArray = [];
+        let offset = 0;
+
+        while (offset !== rawTx.length) {
+            const maxChunkSize = offset === 0 ? MAX_CHUNK_SIZE - 1 - bipPath.length * 4 : MAX_CHUNK_SIZE;
+            const chunkSize = offset + maxChunkSize > rawTx.length ? rawTx.length - offset : maxChunkSize;
+            // APDU fields configuration
+            const apdu = {
+                cla: CLA_FIELD,
+                ins: TX_INS_FIELD,
+                p1: offset === 0 ? (chunkSize < maxChunkSize ? 0x00 : 0x80) : chunkSize < maxChunkSize ? 0x01 : 0x81,
+                p2: curveMask | (chainCode ? 0x01 : 0x00),
```

Comment:
isn't `chainCode` always false? which means that this term will always evaluate to 0x00?

Reply:
Okay, i'm going update to remove the hard code value, but will change to a parameter cause chain code may be require for verification later